src/cpp/server/auth/ServerAuthHandler.cpp

```diff
@@ -783,95 +775,50 @@ Error isUserLicensed(const system::User& user,
       // (by assigning them as an active user), or we've run out of spaces for the
       // license and auth should fail
       size_t numActiveUsers = 0;
-      error = getNumActiveUsers(connection, &numActiveUsers);
+      std::tie(numActiveUsers, error) = overlay::getActiveUserCount(connection);
       if (error)
          return error;
 
-      if ((userLimit > 0) && (numActiveUsers > userLimit))
+      if ((userLimit > 0) && (numActiveUsers >= userLimit))
       {
          // no more space for this user, so don't let them sign in
          *pLicensed = false;
          return Success();
       }
       
-      addUserToDatabase(connection, user, isAdmin);
+      error = addUserToDatabase(connection, user, isAdmin);
+      if (error)
+         return error;
+
+      if(isSigningIn)
+      {
+         // If the user is signing in, we need to update their last sign-in time
+         error = updateLastSignin(connection, user);
```

Comment:
It also doesn't look like this error is returned. Is there a reason we assign it?